src/Symfony/Component/Serializer/Exception/PartialDenormalizationException.php

```diff
@@ -16,22 +16,51 @@
  */
 class PartialDenormalizationException extends UnexpectedValueException
 {
-    private $data;
-    private $errors;
+    private ?ExtraAttributesException $extraAttributesError = null;
 
-    public function __construct($data, array $errors)
+    public function __construct(
+        private $data,
+        /**
+         * @var NotNormalizableValueException[]
+         */
+        private array $notNormalizableErrors,
+        array $extraAttributesErrors = []
+    )
     {
         $this->data = $data;
-        $this->errors = $errors;
+        $this->notNormalizableErrors = $notNormalizableErrors;
+        $extraAttributes = [];
+        foreach ($extraAttributesErrors as $error) {
+            $extraAttributes = \array_merge($extraAttributes, $error->getExtraAttributes());
+        }
+        if (\count($extraAttributes) > 0) {
+            $this->extraAttributesError = new ExtraAttributesException($extraAttributes);
+        }
     }
 
     public function getData()
     {
         return $this->data;
     }
 
+    /**
+     * @deprecated since Symfony 6.2, use getNotNormalizableValueErrors() instead.
+     */
     public function getErrors(): array
     {
-        return $this->errors;
+        return $this->getNotNormalizableValueErrors();
```

Comment:
the method should throw a runtime deprecation
can you remind me why we deprecated the method?
deprecating always comes with a cost so it needs a good enough reason

Reply:
Well this was because it might be used. Should I completely remove it / throw an exception instead of deprecating it?